### PR TITLE
Enable use of Volatile<T> on arm64 to get correct memory consistency

### PR DESCRIPTION
**Pull request by rahku**

### CODE REVIEW

**Comment by rahku** [comment]
PTAL @dotnet/arm64-contrib

**Comment by rahku** [comment]
Thanks to @billwert & @ChrisAhna for finding the issue & providing the fix.

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 checked

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT arm64 release

File: src/inc/volatile.h [hunk 1]
```diff
@@ -479,7 +479,7 @@ class VolatilePtr : public Volatile<P>
 #else
 
 // Disable use of Volatile<T> for GC/HandleTable code except on platforms where it's absolutely necessary.
-#if defined(_MSC_VER) && !defined(_ARM_)
+#if defined(_MSC_VER) && !defined(_ARM_) && !defined(_ARM64_)
```
Comment (gkhanna79) [inline src/inc/volatile.h]: What was the problem description?

Comment (rahku) [inline src/inc/volatile.h]: This issue has been causing frequent deadlocks under server GC, all of which were related to the GC's t_join algorithm breaking down due to writes to volatile join_structure fields being observed out of order on remote threads.

**Pull request merged**